__init__.py

```diff
@@ -3,47 +3,84 @@
 """List and open JetBrains IDE projects."""
 
 import os
-import time
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Union
 from shutil import which
 from xml.etree import ElementTree
-
 from albert import *
 
 md_iid = "0.5"
-md_version = "0.5"
+md_version = "1.0"
 md_name = "Jetbrains"
 md_description = "Open Jetbrains IDE projects"
 md_license = "GPL-3"
 md_url = "https://github.com/mqus/jetbrains-albert-plugin"
 md_maintainers = ["@mqus", "@tomsquest"]
-md_authors = "@mqus"
-
-default_icon = os.path.dirname(__file__) + "/jetbrains.svg"
-HOME_DIR = os.environ["HOME"]
-
-JETBRAINS_XDG_CONFIG_DIR = os.path.join(HOME_DIR, ".config/JetBrains")
-GOOGLE_XDG_CONFIG_DIR = os.path.join(HOME_DIR, ".config/Google")
-
-# for all new (2020.1+) config folders and IntelliJIdea and AndroidStudio, this is the right path.
-NEW_RELATIVE_CONFIG_PATH = "options/recentProjects.xml"
-# for older config folders of other IDEs, the following is right:
-OLD_RELATIVE_CONFIG_PATH = "options/recentProjectDirectories.xml"
+md_authors = ["@mqus", "@tomsquest"]
+
+
+@dataclass
+class Project:
+    name: str
+    path: str
+    last_opened: int
+
+
+@dataclass
+class Editor:
```

Comment:
This is great! Much less hacky than before.

Reply:
Thanks!